### DISCUSSION TITLE
How can I await in a dialog before it closes?

### DISCUSSION THREAD

**Discussion by carlosriveroib** [radix-ui/primitives#1328] [Help]
I want to perform an async action when I press accept in a AlertDialog/Dialog component. But when I press accept the dialog closes. How can I await until the action finish to close the dialog?

Thank you very much

**Comment by benoitgrelard** [comment]
Hey @carlosriveroib, have a look here: https://github.com/radix-ui/primitives/discussions/991#discussioncomment-1682160

**Reply by carlosriveroib** [reply]
Oh, thank you very much. I tried to find that but I couldn't. What about adding that to doc?

**Reply by benoitgrelard** [reply]
That's a great suggestion! I'll add a ticket.

**Reply by benoitgrelard** [reply]
https://github.com/radix-ui/website/issues/374
